### PR TITLE
Extract PrimaryKeyQueryable from Queryable

### PR DESCRIPTION
**Pull request by matthewmcgarvey**
Related to #453 

If we want to support database views in the future, one step in getting there is removing the requirement of the database table having a primary key.

When the time comes to allow not setting a primary key, the `BaseQueryTemplate` not include `PrimaryKeyQueryable` which will remove any ability to query by the primary key for that model.

### CODE REVIEW

**Review by paulcsmith — APPROVED** [review]
Smart! I like this approach

**Comment by paulcsmith** [comment]
Looks like there are some conflicts. Ping me when you're ready and I'll take another look + merge

**Comment by matthewmcgarvey** [comment]
@paulcsmith This is ready to be reviewed again. I did tweak it so that `PrimaryKeyQueryable` does not include `Queryable`. Seemed like the wrong choice to do that inheritance type structure.

File: src/avram/primary_key_queryable.cr [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require "./errors"
+
+module Avram::PrimaryKeyQueryable(T)
+  abstract def id
+  abstract def id(id_val)
+  abstract def query
+
```
Comment (paulcsmith) [inline src/avram/primary_key_queryable.cr:7]: I think these should be removed since the module itself implements the methods. 
```suggestion
```

**Review by paulcsmith — COMMENTED** [review]
I like this new approach of having them implement just the methods they own and including both modules 👍 

Just one thought about removing the abstract def

File: src/avram/primary_key_queryable.cr (hunk 1 above)
Comment (matthewmcgarvey) [reply src/avram/primary_key_queryable.cr]: @paulcsmith done

**Comment by jwoertink** [comment]
@matthewmcgarvey is there anything else you need to do on this? If not, I think it's ready to merge in.

**Review by jwoertink — APPROVED** [review]

**Comment by matthewmcgarvey** [comment]
@jwoertink it's ready, thanks

**Pull request merged**